server/app/views/components/FieldWithLabel.java

```diff
@@ -221,6 +227,18 @@ public FieldWithLabel setScreenReaderText(String screenReaderText) {
     return this;
   }
 
+  /** Set the list of HTML tag IDs that should be used for a11y descriptions. */
+  public FieldWithLabel setAriaDescribedByIds(List<String> ariaDescribedByIds) {
+    this.ariaDescribedByIds = new ArrayList<>(ariaDescribedByIds);
+    return this;
+  }
+
+  /** Set whether the question this field belongs to has errors. */
+  public FieldWithLabel setHasQuestionErrors(boolean hasQuestionErrors) {
```

Comment:
FieldWithLabel is used for rendering fields outside of the context of an application (e.g. within the Admin UI when defining question / program types). Putting questions-specific logic feels like it might be introducing a bit too much coupling with the applicant experience.

What about something like `forceIsInvalid()` to ensure that `aria-invalid` is set even if there aren't any field-specific validation errors?